snowshu/adapters/base_sql_adapter.py

```diff
@@ -19,7 +20,7 @@
 logger = logging.getLogger(__name__)
 
 
-class BaseSQLAdapter:
+class BaseSQLAdapter(ABC):
```

Comment:
why don't to mark all abstracted methods with the @abc.abstractmethod decorator, document the expected behaviour and eliminate the explicit `NotImplementedError` exception?
https://docs.python.org/3/library/abc.html#abc.abstractmethod

Reply:
We can actually do that, fair point. We just need to be aware, that we'll have to be explicit and apply these methods in each or add NotImplementedError() if not applicable to every child class. I'd offer to merge this PR without this refactor, I'll make a separate one after this one is approved to avoid growing this PR more and more, is that ok for you?

Reply:
As far as I am aware, the explicit invocation of NotImplementedError() is an outdated approach. I have noticed that you have partially transitioned to a different approach, which is why I have raised this concern in order to maintain consistency within the project.